src/ts/sign.ts

```diff
@@ -0,0 +1,120 @@
+import { BytesLike, ethers, Signer } from "ethers";
+
+import { hashOrder, Order, ORDER_TYPE_FIELDS, timestamp } from "./order";
+import { isTypedDataSigner, TypedDataDomain } from "./types/ethers";
+
+/**
+ * The signing scheme used to sign the order.
+ */
+export const enum SigningScheme {
+  /**
+   * The EIP-712 typed data signing scheme. This is the preferred scheme as it
+   * provides more infomation to wallets performing the signature on the data
+   * being signed.
+   *
+   * <https://github.com/ethereum/EIPs/blob/master/EIPS/eip-712.md#definition-of-domainseparator>
+   */
+  ERC712,
+  /**
+   * Message signed using eth_sign RPC call.
+   */
+  ETHSIGN,
+}
+
+/**
+ * The signature of an order.
+ * It includes the information needed to determine which signing scheme is used.
+ */
+export interface Signature {
+  /**
+   * The signing scheme used in the signature.
+   */
+  scheme: SigningScheme;
+  /**
+   * The signature bytes as prescribed by the selected signing scheme.
+   */
+  digest: BytesLike;
```

Comment:
I typically conflate "digest" with "hash" (maybe my definitions are wrong), and this isn't a hash. Should this be called `data` (although that is super generic name)?